src/CoreBundle/DataModel/BackendTreeNodeDataModel.php

```diff
@@ -0,0 +1,283 @@
+     */
+    public function addChildren(BackendTreeNodeDataModel $treeNodeDataModel): void
+    {
+        $this->children[] = $treeNodeDataModel;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isChildrenAjaxLoad(): bool
+    {
+        return $this->childrenAjaxLoad;
+    }
+
+    /**
+     * @param bool $childrenAjaxLoad
+     */
+    public function setChildrenAjaxLoad(bool $childrenAjaxLoad): void
+    {
+        $this->childrenAjaxLoad = $childrenAjaxLoad;
+    }
+
+    /**
+     * @return string
+     */
+    public function getType(): string
+    {
+        return $this->type;
+    }
+
+    /**
+     * @param string $type
+     */
+    public function setType(string $type): void
+    {
+        $this->type = $type;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isSelected(): bool
+    {
+        return $this->selected;
+    }
+
+    /**
+     * @param bool $selected
+     */
+    public function setSelected(bool $selected): void
+    {
+        $this->selected = $selected;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isDisabled(): bool
+    {
+        return $this->disabled;
+    }
+
+    /**
+     * @param bool $disabled
+     */
+    public function setDisabled(bool $disabled): void
+    {
+        $this->disabled = $disabled;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isOpened(): bool
+    {
+        return $this->opened;
+    }
+
+    /**
+     * @param bool $opened
+     */
+    public function setOpened(bool $opened): void
+    {
+        $this->opened = $opened;
+    }
+
+    /**
+     * @return array
+     */
+    public function getLiAttr(): array
+    {
+        return $this->liAttr;
+    }
+
+    /**
+     * @param array $liAttr
+     */
+    public function setLiAttr(array $liAttr): void
+    {
+        $this->liAttr = $liAttr;
+    }
+
+    /**
+     * @return array
+     */
+    public function getAAttr(): array
+    {
+        return $this->aAttr;
+    }
+
+    /**
+     * @param array $aAttr
+     */
+    public function setAAttr(array $aAttr): void
+    {
+        $this->aAttr = $aAttr;
+    }
+
+    /**
+     * @return string
+     */
+    public function getConnectedPageId(): string
+    {
+        return $this->connectedPageId;
+    }
+
+
+    /**
+     * Specify data which should be serialized to JSON
+     * @link https://php.net/manual/en/jsonserializable.jsonserialize.php
+     * @return mixed data which can be serialized by json_encode,
+     * which is a value of any type other than a resource.
+     * @since 5.4.0
+     */
+    public function jsonSerialize()
+    {
+        $jsTreeItem = [
+            'id' => $this->id,
+            'text' => $this->name,
+            'type' => $this->type,
+            'state' => [
+                'opened' => $this->opened,
+                'disabled' => $this->disabled,
+                'selected' => $this->selected
+            ],
+            'li_attr' => $this->liAttr,
```

Comment:
if the liAttr is renamed. The keys should change here, too. This will need changes in the JS, too.